applications/nrf_desktop/CMakeLists.txt

```diff
@@ -47,6 +47,62 @@ if(CONFIG_IMG_MANAGER)
   zephyr_library_link_libraries(MCUBOOT_BOOTUTIL)
 endif()
 
+if(CONFIG_BOOTLOADER_MCUBOOT)
+  if(NOT (CONFIG_MCUBOOT_BOOTLOADER_MODE_SINGLE_APP OR
+          CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP OR
+          CONFIG_MCUBOOT_BOOTLOADER_MODE_RAM_LOAD OR
+          CONFIG_MCUBOOT_BOOTLOADER_MODE_SWAP_USING_MOVE))
+    message(WARNING "
+            ------------------------------------------------------------------------------
+            --- WARNING: The currently selected mode of the MCUboot bootloader has not ---
+            --- been used or tested with any nRF Desktop configuration.                ---
+            ------------------------------------------------------------------------------
+            ")
+  endif()
+
+  if(CONFIG_MCUBOOT_BOOTLOADER_MODE_RAM_LOAD)
+    message(WARNING "
+            -----------------------------------------------------------------------------
+            --- WARNING: The RAM load mode of the MCUboot bootloader is experimental. ---
+            -----------------------------------------------------------------------------
+            ")
+
+    # The RAM load requires the code partition to be defined in DTS.
+    if(CONFIG_PARTITION_MANAGER_ENABLED OR (NOT CONFIG_USE_DT_CODE_PARTITION))
+      message(FATAL_ERROR
+              "The RAM load mode of the MCUboot bootloader is not supported with the Partition "
+              "Manager (PM). Please disable PM and use DTS for defining the memory layout.")
```

Comment:
Consider mentioning about enabling `CONFIG_USE_DT_CODE_PARTITION`. Otherwise the displayed message might be a bit misleading in case `CONFIG_PARTITION_MANAGER_ENABLED=n` + `CONFIG_USE_DT_CODE_PARTITION=n`